Client/Frontend/Browser/TabMetadataManager.swift

```diff
@@ -20,7 +20,7 @@ class TabMetadataManager {
         tabGroupData.tabHistoryCurrentState == TabGroupTimerState.openURLOnly.rawValue
     }
 
-    init(profile: Profile) {
+    init(profile: Profile = AppContainer.shared.resolve(type: Profile.self) as! Profile) {
```

Comment:
When I made my previous comment I didn't really think  it through, I thought we'd be able to drop the `as Profile` part, but since it returns an Any object that's obviously impossible.
I believe there should be a way of having our resolve method use generics in order to avoid having to force cast every time we use it. If that is something you want to take a crack at then go for it but if you have never worked with generics before it can be pretty intimidating so I'd be happy to pick it up when I get back or work with you on it if it's something you want to learn. 

It's not a blocker, it's a minor refactor that can happen at a later stage.